### DISCUSSION TITLE
How to create 6 hourly aggregation

### DISCUSSION THREAD

**Discussion by obones** [open-meteo/open-meteo#664] [Q&A]
Hello,

I'm starting to work with the Forecast API and am trying to figure out how to create a 6 hourly aggregation to show the coming weather by quarter of a day (morning, afternoon, evening, night)

I read the documentation for the API and I did not see a ready made solution for this, but that doesn't mean there isn't a way to construct it from the hourly results.

What would you recommend that I look at to get me started?

**Comment by obones** [comment]
What's strange is that the Java SDK offers a `sixHourly` method on `WeatherAPIResponse` as can be seen [here](https://github.com/open-meteo/sdk/blob/main/java/com/openmeteo/sdk/WeatherApiResponse.java#L53)

But it stays empty no matter what I tried.

**Reply by patrick-zippenfenig** [reply]
Hi, six-hourly data is not yet available in any "finished" API. There is a work-in-progress API for seasonal forecast data which contains 6 hourly data: https://open-meteo.com/en/docs/seasonal-forecast-api

6 hourly data aggregations may be added to the regular weather API at some point (no fixed plans yet)

**Reply by obones** [reply]
Thanks for the confirmation, I understand that there are priorities to be decided.

If I was to create such an aggregation myself in my language of choice from the hourly forecast, what resources would you recommend that I read? 
For instance, I'm not sure how to compute the dominant wind from 6 different forecast.

**Reply by patrick-zippenfenig** [reply]
For most variables like temperature maximum or minimum it should be straight forward. For dominant wind direction I use vector addition. Simply convert wind speed and direction to U/V vectors, add them for every hour and afterwards derive wind direction again.
